app/services/schools/training_programme.rb

```diff
@@ -40,7 +40,7 @@ def mentors_at_school?
     def ect_training_programme
       return school.transient_ects_at_school_training_programme if school.respond_to?(:transient_ects_at_school_training_programme)
 
-      ECTAtSchoolPeriod.where(id: ects_expressions_of_interest.pluck(:id) + ect_at_school_periods.pluck(:id)).order(training_programme: :asc).pick(:training_programme)
+      ECTAtSchoolPeriod.joins(:training_periods).where(id: ects_expressions_of_interest.pluck(:id) + ect_at_school_periods.pluck(:id)).order(training_programme: :asc).pick('training_periods.training_programme')
```

Comment:
Also this is probably cleaner coming from the `TrainingPeriod` point of view now


```suggestion
      TrainingPeriod
        .joins(:ect_at_school_period)
        .where(ect_at_school_period: { id: ects_expressions_of_interest.pluck(:id) + ect_at_school_periods.pluck(:id) })
        .order(training_programme: :asc)
        .pick(:training_programme)
```

Reply:
With no joins maybe better?

```
TrainingPeriod.where(ect_at_school_period_id: ects_expressions_of_interest.pluck(:id) + ect_at_school_periods.pluck(:id)).order(training_programme: :asc).pick(:training_programme)
```